docs/topics/compatibility-guide-17.md

```diff
@@ -0,0 +1,502 @@
+
+### Type nullability enhancement improvements
+
+> **Issue**: [KT-48623](https://youtrack.jetbrains.com/issue/KT-48623)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.7 will change how it loads and interprets type nullability annotations in Java code
+>
+> **Deprecation cycle**:
+>
+> - 1.4.30: introduce warnings for cases where more precise type nullability could lead to an error
+> - 1.7.0: infer more precise nullability of Java types,
+>   `-XXLanguage:-TypeEnhancementImprovementsInStrictMode` can be used to temporarily revert to the pre-1.7 behavior
+
+### Prevent implicit coercions between different numeric types
+
+> **Issue**: [KT-48645](https://youtrack.jetbrains.com/issue/KT-48645)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: Kotlin will avoid converting numeric values automatically to a primitive numeric type where only a downcast to that type was needed semantically
+>
+> **Deprecation cycle**:
+>
+> - < 1.5.30: the old behavior in all affected cases
+> - 1.5.30: fix the downcast behavior in generated property delegate accessors,
+>   `-Xuse-old-backend` can be used to temporarily revert to the pre-1.5.30 fix behavior
+> - \>= 1.7.20: fix the downcast behavior in other affected cases
+
+### Deprecate the `enabled` and the `compatibility` modes of the compiler option -Xjvm-default
+
+> **Issue**: [KT-46329](https://youtrack.jetbrains.com/issue/KT-46329)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.6.20 warns about the usage of `enabled` and `compatibility` modes of the `-Xjvm-default` compiler option
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: introduce a warning on the `enabled` and `compatibility` modes of the `-Xjvm-default` compiler option
+> - \>= 1.8.0: raise this warning to an error
+
+### Prohibit calls to functions named `suspend` with a trailing lambda
+
+> **Issue**: [KT-22562](https://youtrack.jetbrains.com/issue/KT-22562)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.6 no longer allows calling user functions named `suspend` that have the single argument of a functional type passed as a trailing lambda
+>
+> **Deprecation cycle**:
+>
+> - 1.3.0: introduce a warning on such function calls
+> - 1.6.0: raise this warning to an error
+> - 1.7.0: introduce changes to the language grammar so that `suspend` before `{` is parsed as a keyword
+
+### Prohibit smart cast on base class property if the base class is from another module
+
+> **Issue**: [KT-52629](https://youtrack.jetbrains.com/issue/KT-52629)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.7 will no longer allow smart casts on properties declared in another module 
+> that were erroneously allowed before
```

Comment:
```suggestion
```

Reply:
The intent here is to show that it was not in all cases of properties in another module, but only in those that were mistakenly allowed

Reply:
Then we need to specify what kind of cases were incorrectly allowed, because now it doesn't make much sense — "Kotlin will no longer allow smarcasts... that were erroneously allowed before"